dockers/docker-fpm-frr/bgpcfgd

```diff
@@ -390,6 +407,96 @@ class BGPPeerMgr(Manager):
         return peers
 
 
+class InterfaceMgr(Manager):
+    def __init__(self, daemon, directory, interface_table = swsscommon.CFG_INTF_TABLE_NAME):
+        super(InterfaceMgr, self).__init__(
+            daemon,
+            directory,
+            [],
+            "CONFIG_DB",
+            interface_table
+        )
+
+    def set_handler(self, key, data):
+        # Interface table can have two keys, 
+        # one with ip prefix and one without ip prefix
+        if '|' in key:
+            data = {}
+            data["interface"], network = key.split('|', 1)
+            try:
+                network = netaddr.IPNetwork(str(network))
+            except:
+                syslog.syslog(syslog.LOG_WARNING, 'Subnet {} format is wrong'.format(network))
+                return False
+            data["prefixlen"] = str(network.prefixlen)
+            ip = str(network.ip)
+            self.directory.put("local_addresses", ip, data)
+        else:
+            self.directory.put("interfaces", key, data)
+        return True
+
+    def del_handler(self, key):
+        if '|' in key:
+            _, network = key.split('|', 1)
+            try:
+                network = netaddr.IPNetwork(str(network))
+            except:
+                syslog.syslog(syslog.LOG_WARNING, 'Subnet {} format is wrong'.format(network))
+                return False
+            ip = str(network.ip)
+            self.directory.remove("local_addresses", ip)
+        else:
+            self.directory.remove("interfaces", key)
+
+    @staticmethod
+    def get_local_interface(directory, local_addr):
+        local_addresses = directory.get_slot("local_addresses")
+        # Check if the local address of this bgp session has been set
+        if local_addr not in local_addresses:
+            return None
+        local_address = local_addresses[local_addr]
+        interfaces = directory.get_slot("interfaces")
+        # Check if the information for the interface of this local address has been set
+        if local_address.has_key("interface") and local_address["interface"] in interfaces:
+            return interfaces[local_address["interface"]]
+        else:
+            return None
+
+    @staticmethod
+    def get_vnet(interface):
+        if interface.has_key("vnet_name") and interface["vnet_name"]:
```

Comment:
description here.

Check here for example.

https://github.com/Azure/sonic-mgmt/blob/master/tests/common/platform/transceiver_utils.py

Reply:
Done it